server/src/internalClusterTest/java/org/elasticsearch/search/fetch/FetchSubPhasePluginIT.java

```diff
@@ -114,7 +115,7 @@ public void testPlugin() throws Exception {
         private static final String NAME = "term_vectors_fetch";
 
         @Override
-        public FetchSubPhaseProcessor getProcessor(SearchContext searchContext) {
+        public FetchSubPhaseProcessor getProcessor(SearchContext searchContext, SearchLookup lookup) {
```

Comment:
I wonder if in all places where we carry around the search lookup, we should not rather make it a supplier like we did in fielddataBuilder.

Reply:
If we did the `Supplier` we could automatically `forkAndTrackReferences` when you call `get` on it which is kind of neat. I like it!

Reply:
@jtibshirani what do you think? I was under the impression that the lookup is not used in many places, and maybe we could / should create it lazily. We do that already in fielddataBuilder, though it is some indirection that may make the code harder to read.

Reply:
Hmm - I don't think we can know from the signature how to fork....

Reply:
Wait wait wait, we don't need to forkAndTrackReferences because getting doc values does that for us. Ignore that. I think we're actually fine without the Supplier then. Because the only thing it'd let us do is lazily build the `SearchLookup` but that class isn't that heavy so I don't think it is worth it. And we need the `SourceLookup` anyway because the fetch phase is pretty centralized around it.

Reply:
ok so what you are saying is that given fetching revolves around source most of the times, and we get source lookup from search lookup, effectively a supplier does not save much. I was trying to figure out what makes this scenario different compared to fielddataBuilder that would justify having a supplier on one hand but not on the other.

Reply:
I think with field data builder we typically don't even need the lookup at all - only really runtime fields want it.

Reply:
Possibly dumb question - is it worth putting it into the SearchContext, rather than having it as an extra parameter here?

Reply:
It is all but a dumb question to me. In fact when I saw the new argument the first thing I did was checking that it was not already available in search context :)

Reply:
I don't want to add it to `SearchContext` because it'd have a *huge* blast radius. I think it'd be reasonable to make a `FetchContext` which has both things. But I don't think it is worth it because there are only 2 arguments..

Reply:
`FetchContext` might be an interesting followup, but sure let's leave it for now.

Reply:
I think if we did `FetchContext` I'd want it to be *smaller* than the `SearchContext`. That thing is just too big....